app/controllers/publishers/paypal_accounts_controller.rb

```diff
@@ -0,0 +1,55 @@
+require "faraday"
+
+module Publishers
+  class PaypalAccountsController < ApplicationController
+    before_action :authenticate_publisher!
+
+    def disconnect
+      current_publisher.paypal_connections.active.update_all(hidden: true)
+    end
+
+    def refresh
+      # (Albert Wang): I don't think Paypal will go down that often, but we can change this into perform_later
+      # if this is a long running call.
+      Paypal::RefreshIdentity.new(publisher_id: current_publisher.id).perform
+      flash[:alert] = I18n.t('publishers.paypal_account_panel.still_not_verified') unless current_publisher.paypal_connection.verified_account?
+      redirect_to home_publishers_path
+    end
+
+    def connect_callback
+      authorization_code = params[:code]
+      result = Faraday.post("#{Rails.application.secrets[:paypal_api_uri]}/v1/oauth2/token") do |req|
```

Comment:
In the future I would like us to extract this into a bare-bones API class that gets created in an initializer so that we can do the following.

```ruby
access_token, refresh_token = PaypalClient.request_token(authorization_code: params[:code])
user_info = PaypalClient.user(access_token: access_token)

paypal_connection = PaypalConnection.find_or_initialize_by(
  user_id: current_publisher.id,
  email: user_info.email
)
paypal_connection.update(
  refresh_token: refresh_token,
  email: user_info.email,
  country: user_info.country,
  verified_account: user_info.verified_account == 'true',
  paypal_account_id: user_info.id,
  hidden: false
)
```

This is fine for now, but in the future let's keep this in mind.